AppCenter/AppCenter/Internals/Channel/MSChannelUnitDefault.m

```diff
@@ -270,6 +246,13 @@ - (void)sendLogContainer:(MSLogContainer *__nonnull)container
                                               [delegate channel:self didFailSendingLog:aLog withError:error];
                                             }
                                           }];
+
+                // Disable and delete all data on fatal error.
+                if([MSHttpUtil isRecoverableError:response.statusCode]) {
+                  MSLogError([MSAppCenter logTag], @"Fatal error encountered; shutting down channel unit with group ID %@", self.configuration.groupId);
+                  [self setEnabled:NO andDeleteDataOnDisabled:YES];
+                  [self resetTimer];
```

Comment:
Should we return here? We don't need to do the rest of code.

Reply:
We need to double check whether `[self resetTimer]` is necessary here.

Reply:
@jaelim-ms `pauseWithIdentifyingObjectSync` already has a `[self resetTimer]` call internally, but I don't know how we know if it is necessary here.